flow/connectors/bigquery/bigquery.go

```diff
@@ -149,12 +149,22 @@ func NewBigQueryConnector(ctx context.Context, config *protos.BigqueryConfig) (*
 		return nil, fmt.Errorf("failed to create BigQueryServiceAccount: %v", err)
 	}
 
+	datasetID := config.GetDatasetId()
+	datasetParts := strings.Split(datasetID, ".")
```

Comment:
```suggestion
	part1, part2, hasParts := strings.Cut(datasetID, ".")
```
Then error `if hasParts && strings.Contains(part2, ".")`